sfge/src/main/java/com/salesforce/rules/UnusedMethodRule.java

```diff
@@ -166,64 +168,66 @@ private boolean directedToSkip(MethodVertex methodVertex) {
-            // The method call has a number of children equal to one more than the
-            // arity of the method being called.
-            List<ChainedVertex> parameters = potentialCall.getParameters();
-            if (parameters.size() != methodVertex.getArity()) {
-                continue;
-            }
-            // If we're at this point, then all of our checks were satisfied,
-            // and this method call seems to be an invocation of our method.
+            // If we're here, then our checks were satisfied, and this method call
+            // appears to be an invocation of the target method.
             return true;
         }
         // If we're at this point, none of the method calls satisfied our checks.
         return false;
     }
 
-    private List<MethodCallExpressionVertex> getInternalCalls(String definingType) {
+    private List<InvocableWithParametersVertex> getInternalCalls(String definingType) {
         // First, check if we've got anything for the desired type.
         // If so, we can just return that.
         if (this.internalMethodCallsByDefiningType.containsKey(definingType)) {
             return this.internalMethodCallsByDefiningType.get(definingType);
         }
         // Otherwise, we'll need to do a query for internal calls.
-        // An internal call is any method call whose reference is `this` or empty.
+        // For instance methods, an internal call is any method call whose reference is `this` or
+        // empty.
         String[] desiredReferenceTypes =
                 new String[] {
                     NodeType.THIS_VARIABLE_EXPRESSION, NodeType.EMPTY_REFERENCE_EXPRESSION
                 };
-        List<MethodCallExpressionVertex> internalCalls =
+        List<InvocableWithParametersVertex> internalCalls =
                 SFVertexFactory.loadVertices(
                         g,
-                        g.V().where(
-                                        H.has(
-                                                NodeType.METHOD_CALL_EXPRESSION,
-                                                Schema.DEFINING_TYPE,
-                                                definingType))
-                                // NOTE: There's no .hasLabel(String...) method, hence the call to
-                                // .hasLabel(String,String...).
-                                .where(
-                                        __.repeat(__.out(Schema.CHILD))
-                                                .until(
-                                                        __.hasLabel(
-                                                                desiredReferenceTypes[0],
-                                                                desiredReferenceTypes))
-                                                .count()
-                                                .is(P.gt(0))));
+                        g.V().or(
+                                        __.where(
```

Comment:
208-225 is the same query, nested inside an `OR` with another query that gets `ThisMethodCallExpression` vertices.